datastore/sqlite_datastore.nim

```diff
@@ -240,33 +245,35 @@ method get*(
     bytes = dataCol(s, 0)
 
   let
-    exists = ? self.getStmt.query((key.id), onData)
+    queryRes = self.getStmt.query((key.id), onData)
+
+  if queryRes.isErr: return failure queryRes.error.msg
 
-  if exists:
-    success bytes.some
+  if queryRes.get:
+    return success bytes.some
   else:
-    success seq[byte].none
+    return success seq[byte].none
```

Comment:
what about 
```Nim
var bytes: ?seq[byte] = none
...
if queryRes.isErr:
  return queryRes
else:
  return success bytes
```
?

Reply:
For that one if we do `if queryRes.isErr: return queryRes` then the compiler fails with:
```
Error: type mismatch: got <Future[Result[Option[seq[byte]], ref CatchableError]], Result[system.bool, ref CatchableError]> ... but expression 'queryRes' is of type: Result[system.bool, ref CatchableError]
```

In any case what we want to return is the seq in `var bytes`, which is populated via side-effect in the query callback.

Reply:
yes, we already discussed isErr part in another thread.

what about the `else` part - instead of checking queryRes.get, just return `bytes` that is initialized with `none`, but can be overwritten in onData

Reply:
Change made: https://github.com/status-im/nim-datastore/pull/15/commits/2a7e1b987e2e8f2687560e06d58ab035432b307f

Thanks for spotting that!